stdlib/public/core/Reverse.swift

```diff
@@ -12,34 +12,47 @@
 
 /// An index that traverses the same positions as an underlying index,
 /// with inverted traversal direction.
-public struct ReverseIndex<Base> : Comparable {
-  public init(_ base: Base) { self.base = base }
+public struct ReverseIndex<Base : BidirectionalCollection> : Comparable {
```

Comment:
Right (almost):

`public struct ReverseIndex<Base : Comparable> ...`

The reason is that we want indices for the collection, its slices, its reverse, and reverse of the slices, slices of the reverse, to be compatible without too much re-wrapping just for type system's sake.